### PR TITLE
Refactor controller: separate order, wallet and inventory controller from user controller

### PR DESCRIPTION
**Pull request by vnayakg**

### CODE REVIEW

File: src/main/kotlin/com/esops/controller/GlobalExceptionController.kt [hunk 1]
```diff
@@ -68,6 +65,11 @@ class GlobalExceptionController {
         return HttpResponse.badRequest(ErrorResponse(error.errorList))
     }
 
+    @Error(global = true)
+    fun cannotAddOrderError(error: CannotAddOrderException): HttpResponse<ErrorResponse> {
```
Comment (AarthiT) [inline src/main/kotlin/com/esops/controller/GlobalExceptionController.kt]: why can't you add? naming please

File: src/main/kotlin/com/esops/service/OrderService.kt [hunk 2]
```diff
@@ -18,6 +18,7 @@ class OrderService(private var userService: UserService,
     private var orderIDCounter: Long = 0
 
     fun placeOrder(username: String, addOrderRequestBody: AddOrderRequestBody): Order {
+        userService.canAddOrder(username, addOrderRequestBody)
```
Comment (AarthiT) [inline src/main/kotlin/com/esops/service/OrderService.kt]: again can?

**Comment by AarthiT** [comment]
does all controller have tests?

**Comment by vnayakg** [comment]
We have organised tests and renamed canOrder function

File: src/main/kotlin/com/esops/service/UserService.kt [hunk 3]
```diff
@@ -133,14 +131,14 @@ class UserService(private var vestingConfiguration: VestingConfiguration,
...
             EsopType.NON_PERFORMANCE -> {
                 val unvestedInventory =
-                        UnvestedInventory(dividedInventory = breakIntoVestingCycles(addInventoryRequestBody.quantity!!))
+                    UnvestedInventory(dividedInventory = breakIntoVestingCycles(addInventoryRequestBody.quantity!!))
                 users[username]!!.unvestedInventoryList.add(unvestedInventory)
                 AddInventoryResponseBody("${addInventoryRequestBody.quantity} ESOPs added to your account")
             }
 
             EsopType.PERFORMANCE -> {
                 users[username]!!.performance.free =
```
Comment (AarthiT) [inline src/main/kotlin/com/esops/service/UserService.kt:140]: please read about nullables

**Pull request merged**